posthog/hogql_queries/utils/query_previous_period_date_range.py

```diff
@@ -15,13 +15,13 @@ class QueryPreviousPeriodDateRange(QueryDateRange):
     """Translation of the raw `date_from` and `date_to` filter values to datetimes."""
 
     _team: Team
-    _date_range: Optional[InsightDateRange]
+    _date_range: Optional[InsightDateRange | DateRange]
```

Comment:
A: They look the same except for defaults
```python
class InsightDateRange(BaseModel):
    model_config = ConfigDict(
        extra="forbid",
    )
    date_from: Optional[str] = "-7d"
    date_to: Optional[str] = None
    explicitDate: Optional[bool] = Field(
        default=False,
        description=(
            "Whether the date_from and date_to should be used verbatim. Disables rounding to the start and end of"
            " period."
        ),
    )

class DateRange(BaseModel):
    model_config = ConfigDict(
        extra="forbid",
    )
    date_from: Optional[str] = None
    date_to: Optional[str] = None
    explicitDate: Optional[bool] = Field(
        default=False,
        description=(
            "Whether the date_from and date_to should be used verbatim. Disables rounding to the start and end of"
            " period."
        ),
    )
```

Reply:
Is there any reason not to use InsightDateRange?

Reply:
Basically because of the default. Insights want to set 7 days as default, we're happy with nothing. I'll change it to have no default to improve the code readability, we can always revert it